### DISCUSSION TITLE
3.0.0-preview.4

### DISCUSSION THREAD

**Discussion by tfiszer** [lepoco/wpfui#804] [Q&A] [answered]
Where's the 3.0.0-preview.4 coming from? There's no tag, branch, nor any reference to it on GitHub, yet I found 3.0.0-preview.8 mentioned in the props files. 

What's up with the versioning?

I also found that 3.0.0-preview.4 version of code created by the WPF UI VS Templates extension, doesn't work when changing the nuget packages down to latest release.

**Comment by lahma0** [comment]
I am also trying to figure this out...

**Comment by pomianowski** [comment, marked as answer]
Preview versions come from the merges to the main branch. The feature of the **git** is that you can go back to any state of the code in the history.

Merge from preview.4 is a commit with hash 35840237797de317d57e6b3be4039db46e13760f
https://github.com/lepoco/wpfui/tree/35840237797de317d57e6b3be4039db46e13760f

When we exit the preview, official releases will appear as tags.
